Wire up FOLIO date_cataloged to instance catalogedDate

thatbudakguy opened the pull request:
Fixes #691

cbeer commented on lib/traject/config/folio_config.rb:
```diff
@@ -146,18 +146,18 @@ def holdings(record, context)
+  case timestamp
+  when Numeric
+    accumulator << Time.at(timestamp / 1000).utc.iso8601
+  when String && (timestamp !~ /NEVER/i)
```
Are we populating `catalogedDate` with `NEVER` in FOLIO, or is that a symphony artifact?

thatbudakguy replied:
in slack, @shelleydoljack said:
> coming out of symphony, it’ll be 0. should we leave it that way? i think we’re converting that to NEVER in our export.

so i think we could do it either way — i opted for this for parity with the sirsi config, but maybe there's something simpler.

thatbudakguy replied:
i think if we don't replace 0 with `NEVER`, the current logic will put it in the `Numeric` bucket and it'll become the 1970 date, which seems less accurate than just not indexing a value there.

shelleydoljack replied:
Coming out of Symphony, when date cataloged is never, the API returns 0. When migrating to FOLIO, this should translate to what FOLIO does, which is not write a value to the instance field catalogedDate. So, for FOLIO, line 150 and 158 matter and Symphony-migrated records in FOLIO won't need line 159. However, because IndexData is taking date cataloged from MARC 008, which could be a string (e.g. "19uu-uu-uu", but that's a migration error, they should not write to that field when data looks like this), then we should account for that in some way, like not adding it to the `date_cataloged` solr field.

shelleydoljack commented on lib/traject/config/folio_config.rb:
```diff
@@ -146,18 +146,18 @@ def holdings(record, context)
-# when are these timestamps updated in FOLIO?
 to_field 'date_cataloged' do |record, accumulator|
-  timestamp = record.record.dig('metadata', 'createdDate')
+  timestamp = record.instance['catalogedDate'] || record.instance.dig('metadata', 'createdDate')
```
It looks like when there is no cataloged data in FOLIO, the field is not returned in the response for something like `GET https://${OKAPI_URL}/inventory/instances?query=hrid=="in00000000224"`. I don't think we should use createdDate if `record.instance['catalogedDate']` is missing because it could put way too many things at the top of the sort when we batch-load records into the system, making "new to the library" sort of useless.

thatbudakguy commented:
fixed this up after pairing with @shelleydoljack. we no longer try to populate the field if `catalogedDate` is empty because it could cause weird behavior for the date facet, and we guard against dates with `u` in them coming out of MARC, which we saw happening for Lane records that IndexData handled.

also added tests.

cbeer commented on lib/traject/config/folio_config.rb:
```diff
@@ -146,22 +146,12 @@ def holdings(record, context)
-    accumulator << Time.iso8601(timestamp).utc.iso8601
-  end
+  timestamp = record.instance['catalogedDate']
+  accumulator << Date.parse(timestamp).to_time.utc.at_beginning_of_day.iso8601 if timestamp && !timestamp.include?('u')
```
Maybe it's better to check for what we want instead of playing whack-a-mole with the stuff we don't?

```suggestion
  accumulator << Time.parse(timestamp).utc.at_beginning_of_day.iso8601 if timestamp =~ /\d{4}-\d{2}-\d{2}/
```
